### PR TITLE
aggregate by all if aggregate is empty

### PR DESCRIPTION
**Pull request by nickcurie**
## What does this PR change?
* 

## Does this PR relate to any other PRs?
* 

## How will this PR impact users?
* 

## Does this PR address any GitHub or Zendesk issues?
* Closes ...

## How was this PR tested?
* 

## Does this PR require changes to documentation?
* 

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next OpenCost release? If not, why not?
*

### CODE REVIEW

File: pkg/cloudcost/queryservice_helper.go [hunk 1]
```diff
@@ -38,6 +38,11 @@ func ParseCloudCostRequest(qp httputil.QueryParams) (*QueryRequest, error) {
 		aggregateBy = append(aggregateBy, prop)
 	}
 
+	// if we're aggregating by nothing (aka `item` on the frontend) then aggregate by all
+	if aggregateBy != nil && len(aggregateBy) < 1 {
```
Comment (michaelmdresser) [inline pkg/cloudcost/queryservice_helper.go]: `GetList()` returns `nil` if the string is empty https://github.com/opencost/opencost/blob/develop/pkg/util/mapper/mapper.go#L440

```suggestion
	if len(aggregateBy) == 0 {
```

should work

Comment (nickcurie) [reply pkg/cloudcost/queryservice_helper.go]: This fails the tests

Comment (nickcurie) [reply pkg/cloudcost/queryservice_helper.go]: `aggregateBy != nil` was what fixed it but probably caused the issue to not get solved

Comment (michaelmdresser) [reply pkg/cloudcost/queryservice_helper.go]: I think we should change the test; behavior isn't working as its supposed to so we're fixing a bug. I'd say that the unit test is wrong.

Comment (nikovacevic) [reply pkg/cloudcost/queryservice_helper.go]: So, we probably need to do the "all" thing and treat both `nil` and `{}` as "do not aggregate." Which might mean that we need a test change? Does that sound right?

Comment (nickcurie) [reply pkg/cloudcost/queryservice_helper.go]: ok let me nuke the test for now just to make sure this solves the problem

Comment (nickcurie) [reply pkg/cloudcost/queryservice_helper.go]: I'm not sure I'm clear on what "do not aggregate" means. Would that just return one line item or something?

Comment (michaelmdresser) [reply pkg/cloudcost/queryservice_helper.go]: By "do not aggregate" I believe niko means "return maximum-resolution lines", the same as a "full group by"

Comment (nickcurie) [reply pkg/cloudcost/queryservice_helper.go]: So what's `all` then? I thought that would be full group by

Comment (michaelmdresser) [reply pkg/cloudcost/queryservice_helper.go]: I believe "all" is supposed to produce a single, fully-totalled, line

**Review by michaelmdresser — APPROVED** [review]

**Pull request merged**